### PR TITLE
port KH to C++

### PR DESCRIPTION
**Pull request by zingale**
## PR checklist

- [ ] test suite needs to be run on this PR
- [ ] this PR will change answers in the test suite to more than roundoff level
- [ ] all newly-added functions have docstrings as per the coding conventions
- [ ] the `CHANGES` file has been updated, if appropriate
- [ ] if appropriate, this change is described in the docs

### CODE REVIEW

**Comment by zingale** [comment]
blocked by #1422

**Comment by zingale** [comment]
test: http://groot.astro.sunysb.edu/Castro/test-suite/gfortran/2020-12-01-005/index.html

**Comment by maxpkatz** [comment]
It's not required but I suggest removing the Prob_nd.F90

**Comment by zingale** [comment]
new test:
http://groot.astro.sunysb.edu/Castro/test-suite/gfortran/2020-12-01-007/KH.html

File: Exec/hydro_tests/KH/problem_initialize_state_data.H [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+    } else if (problem::problem == 5) {
+
+        dens = problem::rho1 + (problem::rho2 - problem::rho1) * 0.5_rt * (std::tanh((yy - y1) / delta_y ) - std::tanh((yy - y2) / delta_y));
+        velx = vel1 * (std::tanh((yy - y1) / delta_y) - std::tanh((yy - y2) / delta_y ) - 1.0_rt);
+        vely += w0 * std::sin(sine_n * M_PI * xx) * (std::exp(-(yy - y1) * (yy - y1) / (sigma * sigma)) + std::exp(-(yy - y2) * (yy - y2) / (sigma * sigma)));
+        dye  = 0.5_rt * (std::tanh((yy - y2) / delta_y) - std::tanh((yy - y1) / delta_y ) + 2.0_rt);
+
+    } else {
+
+        amrex::Error("Error: This problem choice is undefined.");
```
Comment (maxpkatz) [inline Exec/hydro_tests/KH/problem_initialize_state_data.H:131]: Needs to be guarded out for CUDA

**Review by maxpkatz — APPROVED** [review]

File: Exec/hydro_tests/KH/problem_initialize_state_data.H (hunk 1 above)
Comment (zingale) [reply Exec/hydro_tests/KH/problem_initialize_state_data.H:131]: done

**Pull request merged**